src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -603,7 +603,7 @@ public boolean isNoType(TypeMirror t) {
     }
 
     public boolean isOrdinaryClass(TypeElement te) {
-        if (isEnum(te) || isInterface(te) || isAnnotationType(te)) {
+        if (isEnum(te) || isInterface(te) || isAnnotationType(te) || isRecord(te)) {
```

Comment:
For your consideration ... thinking aloud, this could be done in a way that is better protected against future changes by
```
if (elementUtils.isClass(te) && te.getKind() != ElementKind.CLASS
        || elementUtils.isInterface(te) && te.getKind() != ElementKind.INTERFACE)
    return false;
} 
```